### DISCUSSION TITLE
How is the logger for MLFlow working without registering?

### DISCUSSION THREAD

**Discussion by dhirajsuvarna** [explosion/spaCy#11300] [Help: Coding & Implementations] [answered]
I have seen the code for logging to MLFlow in the [github](https://github.com/explosion/spacy-loggers/blob/main/spacy_loggers/mlflow.py) and I see there is no code to register. 

But in the here in the [doc](https://spacy.io/usage/training#custom-logging) its mentioned that we need to add this line - `@spacy.registry.loggers("my_custom_logger.v1")` to register the custom function.

Can someone explain how this works?

**Comment by polm** [comment, marked as answer]
The decorator is registering an "entry point" - it's an easy to understand and simple way to do that, but there are actually a couple of other ways you could do it. In official spaCy packages we sometimes specify entry points via `setup.cfg` instead, which is what we do for `spacy-loggers`. You can see the source here:

https://github.com/explosion/spacy-loggers/blob/main/setup.cfg

**Comment by dhirajsuvarna** [comment]
Currently spacy-logger loggging in mlflow doesn't support tracking server, however, mlflow is mostly used in a collaborative enviornment, it would be good, if we add the ability to provide tracking server uri from the config file. 
I have done it for my project, can you let me know how can I contribute this to the main repository?

**Reply by polm** [reply]
That sounds like a reasonable feature, if you could submit a PR we'd be glad to take a look at it.
